alacritty/src/renderer/text/builtin_font.rs

```diff
@@ -960,6 +883,39 @@ impl Canvas {
     }
 }
 
+/// Size of the stroke to use.
+fn calculate_stroke_size(width: usize) -> usize {
+    // Use one eight of the cell width, since this is used as a step size for block elements.
+    cmp::max((width as f32 / 8.).round() as usize, 1)
+}
+
+/// Equation for `f(x) = slope * x + offset.
+struct LineEquation {
+    slope: i32,
+    offset: i32,
+    x: i32,
+    width: i32,
+}
+
+impl LineEquation {
+    fn new(slope: i32, offset: i32, from_x: i32, width: i32) -> Self {
+        Self { x: from_x - 1, width, slope, offset }
+    }
+}
+
+impl Iterator for LineEquation {
+    type Item = (f32, f32);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.x += 1;
+        if self.x >= self.width {
+            None
+        } else {
+            Some((self.x as f32, (self.slope * self.x + self.offset) as f32))
+        }
+    }
+}
```

Comment:
Does having this as an iterator actually provide any value over defining it as a function and just iterating over a range? X is just incrementing by `1` anyway, so this is basically the same as `0..width.map(fun)`, isn't it?

Reply:
Yeah, because at some point I want inner functions to return `None` and `unwrap_or` on them? It's just _way_ simpler that way otherwise you need to build up more complex logic.

Reply:
e.g.

```rust
            let p3 = top_inner_line.next().unwrap_or(p2);
            let p4 = bottom_inner_line.next().unwrap_or(p1);
```

Reply:
But a mapped range iterator would produce the exact same result?

Reply:
Hm, let me try to see how it looks.